core/uploads-manager.php

```diff
@@ -0,0 +1,373 @@
+	}
+
+	/**
+	 * Create Temp File
+	 *
+	 * @since 3.2.0
+	 *
+	 * @param string $file_content
+	 * @param string $file_name
+	 * @return string|\WP_Error
+	 */
+	public function create_temp_file( $file_content, $file_name ) {
+		$extension = pathinfo( $file_name, PATHINFO_EXTENSION );
+		// Get the directory for temporary Elementor uploads.
+		$temp_path = $this->get_temp_dir();
+		// Create a random temporary file name.
+		$temp_filename = $temp_path . uniqid() . '.' . $extension;
+
+		if ( 'json' === $extension ) {
+			$file_content = wp_json_encode( $file_content );
+
+			// If the JSON content is not json-encodable, it is not valid.
+			if ( ! $file_content ) {
+				return new \WP_Error( 'Invalid file contents' );
+			}
+		}
+
+		// Save the contents to a temporary file.
+		file_put_contents( $temp_filename, $file_content ); // phpcs:ignore
+
+		return $temp_filename;
+	}
+
+	/**
+	 * Get Temp Directory
+	 *
+	 * Get the temporary files directory path. If the directory does not exist, this method creates it.
+	 *
+	 * @since 3.2.0
+	 *
+	 * @return string $temp_dir
+	 */
+	public function get_temp_dir() {
+		$wp_upload_dir = wp_upload_dir();
+		$tmp_dir = $wp_upload_dir['basedir'] . '/elementor/tmp';
+
+		if ( ! file_exists( $tmp_dir ) && ! is_dir( $tmp_dir ) ) {
+			wp_mkdir_p( $tmp_dir );
+		}
+
+		return $tmp_dir;
+	}
+
+	/**
+	 * Are Unfiltered Uploads Enabled
+	 *
+	 * Checks if the user allowed uploading unfiltered files.
+	 *
+	 * @return bool
+	 * @since 3.2.0
+	 */
+	private function are_unfiltered_uploads_enabled() {
+		if ( ! $this->are_unfiltered_files_enabled ) {
+			$this->are_unfiltered_files_enabled = ! ! get_option( self::UNFILTERED_FILE_UPLOADS_KEY );
+		}
+
+		return $this->are_unfiltered_files_enabled;
+	}
+
+	/**
+	 * Add Mime Type To Allowed Mimes List
+	 *
+	 * @param string $file_type
+	 * @since 3.2.0
+	 */
+	private function add_mime_type_to_allowed_mimes_list( $file_type ) {
+		$file_handler = $this->file_type_handlers[ $file_type ];
+
+		$mime_types = $file_handler->get_mime_types();
+
+		$file_extension = $file_handler->get_file_extension();
+
+		$this->allowed_mime_types[ $file_extension ] = $mime_types;
+	}
+
+	/**
+	 * Save Base64 as File
+	 *
+	 * Saves a Base64 string as a .tmp file in Elementor's temporary files directory.
+	 *
+	 * @param $file
+	 * @return array|\WP_Error
+	 */
+	private function save_base64_to_tmp_file( $file ) {
+		$file_content = base64_decode( $file['fileData'] ); // phpcs:ignore
+
+		// If the decode fails
+		if ( ! $file_content ) {
+			return new \WP_Error( 'file_error', self::INVALID_FILE_CONTENT );
+		}
+
+		$temp_filename = $this->create_temp_file( $file_content, $file['fileName'] );
+
+		if ( is_wp_error( $temp_filename ) ) {
+			return $temp_filename;
+		}
+
+		$new_file_array = [
+			'name' => $file['fileName'],
+			'tmp_name' => $temp_filename,
+			'type' => mime_content_type( $temp_filename ),
+		];
+
+
+		if ( isset( $file['allowedFileTypes'] ) ) {
```

Comment:
Move the ability to set allowed files to the `handle_elementor_upload()` method

Reply:
Done.